src/main/java/io/cdap/plugin/WindowAggregation.java

```diff
@@ -240,24 +305,142 @@ private String generateFrameDescription(WindowAggregationConfig config) {
+      preceding = config.getFrameDefinitionPrecedingBound() + "";
+      following = config.getFrameDefinitionFollowingBound() + "";
+      if (windowFrameType == WindowAggregationConfig.WindowFrameType.ROW) {
+          windowFrameTypeValue = WindowAggregationDefinition.WindowFrameType.ROW;
+      } else if (windowFrameType == WindowAggregationConfig.WindowFrameType.RANGE) {
+          windowFrameTypeValue = WindowAggregationDefinition.WindowFrameType.RANGE;
+      }
+    }
+    List<WindowAggregationConfig.FunctionInfo> functionInfos = config.getAggregates(failureCollector);
+    List<Expression> partitionExpressions = new ArrayList<>(partitionFields.size());
+    List<WindowAggregationDefinition.OrderByExpression> orderExpressions = new ArrayList<>();
+    Map<String, Expression> selectExpressions = new HashMap<>();
+
+    for (String field : partitionFields) {
+      String columnName = getColumnName(expressionFactory, relation, field);
+      Expression partitionExpression = expressionFactory.compile(columnName);
+      partitionExpressions.add(partitionExpression);
+    }
+
+    for (Schema.Field field : Objects.requireNonNull(ctx.getOutputSchema().getFields())) {
+      selectExpressions.put(field.getName(), expressionFactory.compile(getColumnName(
+              expressionFactory, relation, field.getName())));
+    }
+
+    if (partitionOrderFields != null) {
+      String[] split = partitionOrderFields.split(":");
+      WindowAggregationDefinition.OrderByExpression orderByExpression =
+              new WindowAggregationDefinition.OrderByExpression(expressionFactory.compile(split[0].trim()),
+                      WindowAggregationDefinition.OrderBy.valueOf(split[1].trim().toUpperCase()));
+      orderExpressions.add(orderByExpression);
+    }
+
+    for (WindowAggregationConfig.FunctionInfo aggregate : functionInfos) {
+      String alias = aggregate.getAlias();
+      String columnName = getColumnName(expressionFactory, relation, aggregate.getFieldName());
+      WindowAggregationConfig.Function function = aggregate.getFunction();
+      // Check if this function is supported in BigQuery.
+      if (functionBQSqlMap.containsKey(function)
+              && expressionFactory.getCapabilities().contains(StandardSQLCapabilities.BIGQUERY)) {
+        String selectSql = String.format(functionBQSqlMap.get(function), columnName);
+        selectExpressions.put(alias, expressionFactory.compile(selectSql));
+        continue;
+      }
+      return null;
+    }
+    WindowAggregationDefinition.Builder builder = WindowAggregationDefinition.builder()
+              .select(selectExpressions)
+              .partition(partitionExpressions)
+              .orderBy(orderExpressions)
+              .windowFrameType(windowFrameTypeValue);
+    if (windowFrameTypeValue != WindowAggregationDefinition.WindowFrameType.NONE) {
+          builder.unboundedPreceding(unboundedPreceding)
+                  .unboundedPreceding(unboundedFollowing)
```

Comment:
I believe this should be `.unboundedFollowing(unboundedFollowing)`.

Reply:
Done